pkg/controller/kfservice/resources/knative/configuration.go

```diff
@@ -58,25 +65,28 @@ func NewConfigurationBuilder(client client.Client, config *v1.ConfigMap) *Config
 }
 
 func (c *ConfigurationBuilder) CreateKnativeConfiguration(name string, metadata metav1.ObjectMeta, modelSpec *v1alpha1.ModelSpec) (*knservingv1alpha1.Configuration, error) {
-	annotations := make(map[string]string)
+	kfsvcAnnotations := make(map[string]string)
+	filteredAnnotations := utils.Filter(metadata.Annotations, ConfigurationAnnotationDisallowedList)
```

Comment:
If we use a list instead of a map, you may need to overload this filter function to filter keys from a map.

Then, would it be possible to implement this as:
```
annotations := utils.Filter(metadata.Annotations, ConfigurationAnnotationDisallowedList)
```
and avoid any checks to len, etc. The filter function could simply return an empty map if metadata.Annotations is nil.